src/assembler.rs

```diff
@@ -68,11 +66,31 @@ fn make_instruction_map() -> HashMap<String, (InstructionType, u8)> {
         ("hor", ebpf::BPF_HOR),
     ];
 
+    let mem_classes = [
+        (
+            "ldx",
+            LoadReg,
+            ebpf::BPF_MEM | ebpf::BPF_LDX,
+            ebpf::BPF_ALU | ebpf::BPF_X,
```

Comment:
Perhaps we should give `BPF_ALU` another name, as it is going to be associated with memory operations.